http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AzureServiceInterp.scala

```diff
@@ -0,0 +1,351 @@
+
+      controlledResourceOpt <- controlledResourceQuery
+        .getResourceTypeForRuntime(runtime.id, WsmResourceType.AzureVm)
+        .transaction
+      azureRuntimeControlledResource <- F.fromOption(
+        controlledResourceOpt,
+        AzureRuntimeControlledResourceNotFoundException(cloudContext, runtimeName, context.traceId)
+      )
+
+      hasPermission <- authProvider.hasPermission(WsmResourceSamResourceId(azureRuntimeControlledResource.resourceId),
+                                                  WsmResourceAction.Write,
+                                                  userInfo)
+
+      _ <- context.span.traverse(s => F.delay(s.addAnnotation("Done auth call for get azure runtime permission")))
+      _ <- if (hasPermission) F.unit
+      else
+        F.raiseError[Unit](
+          RuntimeNotFoundException(cloudContext, runtimeName, "permission denied", Some(context.traceId))
+        )
+
+    } yield runtime
+
+  override def updateRuntime(userInfo: UserInfo,
+                             runtimeName: RuntimeName,
+                             workspaceId: WorkspaceId,
+                             req: UpdateAzureRuntimeRequest)(implicit as: Ask[F, AppContext]): F[Unit] =
+    F.pure(AzureUnimplementedException("patch not implemented yet"))
+
+  override def deleteRuntime(userInfo: UserInfo, runtimeName: RuntimeName, workspaceId: WorkspaceId)(
+    implicit as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      context <- as.ask
+      azureContext <- wsmDao.getWorkspace(workspaceId).map(_.azureContext)
+      cloudContext = CloudContext.Azure(azureContext)
+
+      runtime <- RuntimeServiceDbQueries.getRuntime(cloudContext, runtimeName).transaction
+      diskId <- runtime.runtimeConfig match {
+        case config: RuntimeConfig.AzureConfig => F.pure(config.persistentDiskId)
+        case _ =>
+          F.raiseError[DiskId](
+            AzureRuntimeHasInvalidRuntimeConfig(cloudContext, runtime.clusterName, context.traceId)
+          )
+      }
+
+      controlledResourceOpt <- controlledResourceQuery
+        .getResourceTypeForRuntime(runtime.id, WsmResourceType.AzureVm)
+        .transaction
+      azureRuntimeControlledResource <- F.fromOption(
+        controlledResourceOpt,
+        AzureRuntimeControlledResourceNotFoundException(cloudContext, runtimeName, context.traceId)
+      )
+
+      hasPermission <- authProvider.hasPermission(WsmResourceSamResourceId(azureRuntimeControlledResource.resourceId),
+                                                  WsmResourceAction.Write,
+                                                  userInfo)
+
+      _ <- context.span.traverse(s => F.delay(s.addAnnotation("Done auth call for delete azure runtime permission")))
+      _ <- if (hasPermission) F.unit
+      else F.raiseError[Unit](RuntimeNotFoundException(cloudContext, runtimeName, "permission denied"))
+
+      _ <- if (runtime.status.isDeletable) F.unit
```

Comment:
minor, but these can be written this way too
```
      _ <- F
        .raiseError[Unit](RuntimeCannotBeDeletedException(cloudContext, runtime.clusterName, runtime.status))
        .whenA(!runtime.status.isDeletable)
```

Reply:
Ah, I do like how this reads better